vernac/vernacentries.ml

```diff
@@ -1536,7 +1536,7 @@ let _ =
       optname  = "kernel term sharing";
       optkey   = ["Kernel"; "Term"; "Sharing"];
       optread  = (fun () -> (Global.typing_flags ()).Declarations.share_reduction);
-      optwrite = (fun b -> Global.set_reduction_sharing b) }
+      optwrite = (fun b -> Global.set_typing_flags { (Global.typing_flags ()) with Declarations.share_reduction = b }) }
```

Comment:
I found the previous code clearer here.

Reply:
It can be moved just before the declare_bool_option though, and at the same time, we could add a `get_reduction_sharing` for the symmetry. It is a bit purely cosmetic but if you insist...

Reply:
I experimented the following:
```ocaml
let get_reduction_sharing () =
  (Global.typing_flags ()).Declarations.share_reduction

let set_reduction_sharing b =
  Global.set_typing_flags
    { (Global.typing_flags ()) with Declarations.share_reduction = b }

let _ =
  declare_bool_option
    { optdepr  = false;
      optname  = "kernel term sharing";
      optkey   = ["Kernel"; "Term"; "Sharing"];
      optread  = (fun () -> (Global.typing_flags ()).Declarations.share_reduction);
      optwrite = (fun b -> Global.set_typing_flags { (Global.typing_flags ()) with Declarations.share_reduction = b }) }
      optread  = get_reduction_sharing;
      optwrite = set_reduction_sharing }
```
I agree that this is clearer, but it is still a bit inconsistent with the rest of the series of `declare_bool_option` in `Vernacentries`. So, I would lean to say that this should go in an independent PR for someone who is willing to make the core more readbable as a whole. What do you think?

Reply:
Maybe we should move this option to Global from Vernacentries. In fact why are random options define in vernacentries instead of next to their bool ref?

Reply:
> In fact why are random options define in vernacentries instead of next to their bool ref?

Iirc, when the `Goptions` mechanism was introduced, most (or all) existing options were put in `Vernacentries`. Then new options arrived and where declared locally to the place they were used. So, yes, it would make sense to now move the options in `Vernacentries` next to where they are relevant.